packages/@coorpacademy-components/src/molecule/setup-cohort-item/index.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import map from 'lodash/fp/map';
+import InputText from '../../atom/input-text';
+import InputCheckbox from '../../atom/input-checkbox';
+import Button from '../../atom/button';
+import style from './style.css';
+
+const InputSplitScreen = props => {
+  const {conditions = {}, groups = {}} = props;
+  const dataCriteria = {
+    groups: {
+      fields: conditions.values
+    }
+  };
+  const dataCollection = {
+    groups: {
+      fields: groups.values
+    }
+  };
+  const criteriaList = map.convert({cap: false})(
+    (criterion, key) => <SetupCohortItem key={key} {...criterion} />,
+    dataCriteria
+  );
+  const collectionList = map.convert({cap: false})(
+    (collection, key) => <SetupCohortItem key={key} {...collection} />,
+    dataCollection
+  );
+  return (
+    <div className={style.split}>
+      <div className={style.splitLeft}>
+        <div>
+          <p className={style.title}>{conditions.title}</p>
+          {criteriaList}
+        </div>
+      </div>
+      <div className={style.splitRight}>
+        <div>
+          <p className={style.title}>{groups.title}</p>
+          {collectionList}
+        </div>
+      </div>
+    </div>
+  );
+};
+
+const ButtonGroup = props => {
+  const {buttonGroup} = props;
+  const data = {
+    groups: {
+      fields: buttonGroup
+    }
+  };
+  const buttons = map.convert({cap: false})(
+    (buttonProps, key) => <SetupCohortItem key={key} {...buttonProps} />,
+    data
+  );
+  return <div className={style.buttonGroup}>{buttons}</div>;
+};
+
+const SetupCohortItem = props => {
+  const {fields = []} = props;
+  const buildInput = field => {
+    const {type} = field;
+    switch (type) {
+      case 'checkbox':
+        return <InputCheckbox {...field} />;
+      case 'splitScreen':
+        return <InputSplitScreen {...field} />;
+      case 'button':
+        return (
+          <div className={style.button}>
+            <Button {...field} />
+          </div>
+        );
+      case 'buttonGroup':
+        return <ButtonGroup {...field} />;
+      default:
+        return <InputText {...field} />;
+    }
+  };
+
+  const buildField = (field, index) => {
+    const input = buildInput(field);
+    return (
+      <div className={style.field} key={index}>
+        {input}
+      </div>
+    );
+  };
+
+  const fieldsList = map.convert({cap: false})(buildField, fields);
+
+  return <div className={style.wrapper}>{fieldsList}</div>;
+};
+
+SetupCohortItem.propTypes = {
+  fields: PropTypes.arrayOf(
```

Comment:
you should not try to reuse the same component as the main component and nested component, it's no easy to read...

```js
SetupCohortItem {
 header :{ 
        title: 'Name',
        value: 'value',
        required: true,
        autoFocus: true
 },
 form: {
   ---> your splitForm props
 }
}
```

then you use 2 different components
1- for the main wrapping
2- for the split

so your proptypes are more accurate and you don't get lose in a recursive rendering.

Reply:
please first ping me when you have your props defined, then when we agree you can go on with the jsx and stuffs

Reply:
Hi @chrisdugne ,
Due to working time, we do not wait for your confirmation when we define props. So I think we will apply and you should check and comment, then we will change if need.

Reply:
actually I've gone deeper in the analysis, and would like even more changes,
I've written a comment: https://github.com/CoorpAcademy/components/pull/1392#issuecomment-442020412